### PR TITLE
refactor(gsaidAPI)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

In order to replace pandas the GSAID API needs tests. this refactoring aims to increase clarity in the code to start creating tests for the functions.

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/upload/gisaid/gisaid.py [hunk 1]
```diff
@@ -47,67 +65,74 @@ def __init__(self, config: CGConfig):
...
 
     def get_gisaid_fasta_path(self, case_id: str) -> Path:
         """Get path to gisaid fasta"""
-        return Path(self.mutant_root_dir, case_id, "results", f"{case_id}.fasta")
+        return Path(self.mutant_root_dir, case_id, "results", f"{case_id}{FileExtensions.FASTA}")
 
     def get_gisaid_csv_path(self, case_id: str) -> Path:
-        """Get path to gisaid csv"""
-        return Path(self.mutant_root_dir, case_id, "results", f"{case_id}.csv")
+        """Get path to gisaid csv."""
```
Comment (henrikstranneheim) [inline cg/meta/upload/gisaid/gisaid.py:104]:
```suggestion
        """Get path to Gisaid CSV."""
```

File: cg/meta/upload/gisaid/gisaid.py [hunk 2]
```diff
@@ -22,10 +22,28 @@
...
+
+class GsaidKeys(StrEnum):
+    COLLECTION_DATE: str = "collection_date"
+    REGION: str = "region"
+    REGION_CODE: str = "region_code"
+    ORIGINAL_LAB: str = "original_lab"
+    ORIGINAL_LAB_ADDRESS: str = "original_lab_address"
+
+
+class GsaidHousekeeperTags(StrEnum):
```
Comment (henrikstranneheim) [inline cg/meta/upload/gisaid/gisaid.py:42]: Perhaps these should go here: cg/constants/housekeeper_tags.py

**Review by henrikstranneheim — APPROVED** [review]
👍

File: cg/meta/upload/gisaid/gisaid.py [hunk 3]
```diff
@@ -22,10 +22,28 @@
 from cg.exc import (
     HousekeeperFileMissingError,
 )
+from cg.constants.constants import FileExtensions, StrEnum
 
 LOG = logging.getLogger(__name__)
 
-UPLOADED_REGEX_MATCH = r"\[\"([A-Za-z0-9_]+)\"\]\}$"
+UPLOADED_REGEX_MATCH: str = r"\[\"([A-Za-z0-9_]+)\"\]\}$"
+KOMPLETTERING: str = "komplettering"
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/meta/upload/gisaid/gisaid.py:30]: I took over the variables that were in the code currently. I am not sure if these come from a file or not. I will alter them if it is possible when establishing the tests.

**Comment by ChrOertlin** [comment]
closing for now - this refactor requires tests to be setup prior. Putting it aside for now in favour of other project

**Pull request closed without merging**